pkg/crc/cache/cache.go

```diff
@@ -0,0 +1,58 @@
+package cache
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/extract"
+	crcos "github.com/code-ready/crc/pkg/os"
+	"github.com/pkg/errors"
+)
+
+type Cache interface {
+	GetBinary(destDir string) (string, error)
```

Comment:
True, `IsCached` is still needed, but can't this be all generic code? I'm not sure an interface is needed, just a type with the appropriate methods should do the trick?

Reply:
This could be squashed in the appropriate commits for example https://github.com/cfergeau/crc/commit/1adb910b1214995e97eff5a532aa8cfea7046ead

Reply:
On second though, `OcCache`, `PodmanCache`, ... are not needed so this can become https://github.com/cfergeau/crc/commit/3b1d32143a1ba749ab8130a1a7503eb737444819

Reply:
@cfergeau we also need to include the path because right now we are dumping everything in `~/.crc/bin` so this will again need a refactor when https://github.com/code-ready/crc/pull/1217 worked on.